Add env:// rendezvous test

pietern opened the pull request:
Summary:
A missing environment variable raised a missing key error. Now it
raises a more descriptive error of the actual problem, for example:

ValueError: env:// rendezvous: WORLD_SIZE not set

Differential Revision: D9888962

soumith commented on torch/distributed/rendezvous.py:
```diff
@@ -101,18 +101,18 @@ def _error(msg):
     if world_size is None:
-        raise _error("world size is missing")
-    rank = os.environ["RANK"]
+        raise _error("WORLD_SIZE not set")
```
nit: once and for all, improving this error message would be nice.
For example, getting this error is much better than before

```
RuntimeError: env:// rendezvous: WORLD_SIZE not set
```

But can be greatly improved:

```
RuntimeError: While initializing torch.distributed using the env:// rendezvous mechanism: environment variable WORLD_SIZE expected, but it is not set.
```

pietern replied:
Good point @soumith, updating.

pietern committed:
Add env:// rendezvous test (#11782)

Summary:
A missing environment variable raised a missing key error. Now it
raises a more descriptive error of the actual problem, for example:

ValueError: Error initializing torch.distributed using env:// rendezvous: environment variable WORLD_SIZE expected, but not set

Pull Request resolved: https://github.com/pytorch/pytorch/pull/11782

Differential Revision: D9888962

fbshipit-source-id: 1134175e30c123043328c9bccff1ad73c9e19ce1
